src/backend/access/transam/xlog.c

```diff
@@ -6936,59 +6675,20 @@ SetLastWrittenLSNForBlockv(const XLogRecPtr *lsns, RelFileLocator relfilenode,
 						   ForkNumber forknum, BlockNumber blockno,
 						   int nblocks)
 {
-	LastWrittenLsnCacheEntry* entry;
-	BufferTag	key;
-	bool		found;
-	XLogRecPtr	max = InvalidXLogRecPtr;
-
-	if (lsns == NULL || nblocks == 0 || lastWrittenLsnCacheSize == 0 ||
-		relfilenode.relNumber == InvalidOid)
-		return InvalidXLogRecPtr;
-
-	key.relNumber = relfilenode.relNumber;
-	key.dbOid = relfilenode.dbOid;
-	key.spcOid = relfilenode.spcOid;
-	key.forkNum = forknum;
-
-	LWLockAcquire(LastWrittenLsnLock, LW_EXCLUSIVE);
-
-	for (int i = 0; i < nblocks; i++)
+	if (set_lwlsn_block_v_hook)
 	{
-		XLogRecPtr	lsn = lsns[i];
-
-		key.blockNum = blockno + i;
-		entry = hash_search(lastWrittenLsnCache, &key, HASH_ENTER, &found);
-		if (found)
-		{
-			if (lsn > entry->lsn)
-				entry->lsn = lsn;
-			else
-				lsn = entry->lsn;
-			/* Unlink from LRU list */
-			dlist_delete(&entry->lru_node);
-		}
-		else
-		{
-			entry->lsn = lsn;
-			if (hash_get_num_entries(lastWrittenLsnCache) > lastWrittenLsnCacheSize)
-			{
-				/* Replace least recently used entry */
-				LastWrittenLsnCacheEntry* victim = dlist_container(LastWrittenLsnCacheEntry, lru_node, dlist_pop_head_node(&XLogCtl->lastWrittenLsnLRU));
-				/* Adjust max LSN for not cached relations/chunks if needed */
-				if (victim->lsn > XLogCtl->maxLastWrittenLsn)
-					XLogCtl->maxLastWrittenLsn = victim->lsn;
-
-				hash_search(lastWrittenLsnCache, victim, HASH_REMOVE, NULL);
-			}
+		return set_lwlsn_block_v_hook(lsns, relfilenode, forknum, blockno, nblocks);
+	}
+	// Behaviour in case the hook is not set. Taken from the implemenatation of SetLastWrittenLSNForBlockv in neon_lwlc.c
```

Comment:
Doesn't make sense to reference our extension here imo. Providing the reasoning inline would be much better

Reply:
This comment still applies.